common/tasks/test-metadata/0.2/test-metadata.yaml

```diff
@@ -69,6 +73,12 @@ spec:
           EVENT_TYPE="pull_request"
         fi
 
+        # The author of pull request is different from the sender for /retest event 
+        # Below change is for konflux update PRs in the /retest event
+        if [[ "$SHA_TITLE" =~ ^chore\(deps\):\ update\ konflux\ references( to .+)?$ ]]; then
+                PR_AUTHOR="red-hat-konflux[bot]"
+        fi
```

Comment:
IMO identifying PR author based on the commit message is not the ideal solution, because this is very specific case, and I'd suggest using something like this instead:
```
PR_AUTHOR=$(curl -s https://api.github.com/repos/${GIT_ORGANIZATION}/${GIT_REPOSITORY}/pulls/${PULL_REQUEST_NUMBER} | jq -r .user.login)
```

So we'd then rely only the GitHub API to provide us the correct data for the PR author, which means that we could omit getting it from `metadata.annotations['pac.test.appstudio.openshift.io/sender']`, which, as you mentioned is not correct in case when someone triggers tests with `/retest` comment.

@flacatus @ascerra wdyt?

Reply:
If we do use this curl command to set the PR_AUTHOR in the script would this also mean that we remove [lines 47-50](https://github.com/konflux-ci/konflux-qe-definitions/pull/66/files#diff-fae681a3e48c1ee7dcf75c1759d50c1ef0d8834d4bf1ccd609eb108520d67b39R47-R50) since we would just be overriding that var anyway